packages/typescript/src/index.ts

```diff
@@ -148,11 +150,9 @@ export default function typescript(options: RollupTypescriptOptions = {}): Plugi
         const tsBuildInfoSource = emittedFiles.get(tsBuildInfoPath);
         // https://github.com/rollup/plugins/issues/681
         if (tsBuildInfoSource) {
-          this.emitFile({
-            type: 'asset',
-            fileName: normalizePath(path.relative(outputOptions.dir!, tsBuildInfoPath)),
-            source: tsBuildInfoSource
-          });
+          const normalizedTsBuildInfoPath = normalizePath(tsBuildInfoPath);
+          fs.mkdirSync(path.dirname(normalizedTsBuildInfoPath), { recursive: true });
```

Comment:
Ideally the plugin should only avoid Rollup's asset system if required. I'd like to see this use `emitFile` if the path falls within the output directory.

- This should use the async versions of the functions and await
- This code should be extracted to a helper function so other parts of the plugin can use it to write outside the dist directory

Reply:
I was working on extracting the code into a helper function outside of `index.ts`, but the implementation feels a bit awkward.

I would need access to `outputOptions`, `parsedOptions`, and `this.emitFile`. It might make more sense to have the helper function inside index.js, but I am unsure if this is okay because all the other functions currently are part of `Plugin`

Reply:
After looking around the repo a bit more, looks like this is quite normal. Working on it